DataFormats/L1TMuon/interface/RegionalMuonShower.h

```diff
@@ -0,0 +1,61 @@
+#ifndef DataFormats_L1Trigger_RegionalMuonShower_h
+#define DataFormats_L1Trigger_RegionalMuonShower_h
+
+#include "DataFormats/Common/interface/Ref.h"
+#include "DataFormats/L1Trigger/interface/BXVector.h"
+#include "DataFormats/L1Trigger/interface/L1TObjComparison.h"
+
+namespace l1t {
+
+  class RegionalMuonShower;
+  typedef BXVector<RegionalMuonShower> RegionalMuonShowerBxCollection;
+  typedef ObjectRefBxCollection<RegionalMuonShower> RegionalMuonShowerRefBxCollection;
+  typedef ObjectRefPair<RegionalMuonShower> RegionalMuonShowerRefPair;
+  typedef ObjectRefPairBxCollection<RegionalMuonShower> RegionalMuonShowerRefPairBxCollection;
+
+  class RegionalMuonShower {
+  public:
+    // Run-3 definitions as provided in DN-20-033
+    // One nominal shower considered higher quality than two loose showers
+    enum Run3Shower { kInvalid = 0, kTwoLoose = 1, kOneNominal = 2 };
+
+    RegionalMuonShower();
+
+    ~RegionalMuonShower();
+
+    bool isValid() const;
+    uint8_t getShowerQualInTime() const { return qualInTime_; }
+    uint8_t getShowerQualOutOfTime() const { return qualOutOfTime_; }
+    bool isTwoLooseInTime() const;
+    bool isOneNominalInTime() const;
+    bool isTwoLooseOutOfTime() const;
+    bool isOneNominalOutOfTime() const;
```

Comment:
@dinyar I think I understand. What I can do is further split up the data into single booleans. 
```
bool isTwoLooseInTime_;
bool isOneNominalInTime_;
bool isTwoLooseOutOfTime_;
bool isOneNominalOutOfTime_;
```
If we were to change the definition, the change would be picked up by CMSSW's data format version control.

I suppose it also gives us the flexibility that a shower is both ```TwoLoose``` and ```OneNominal```, which is not forbidden. It can happen that an even has at least one nominal shower and separately two loose showers, or more than two nominal showers. If you are okay with this, I can update the PR accordingly.

Reply:
Hi @dildick,

Right, however I don't think this helps you when unpacking old RAW data. As requested by @rekovic I've opened a ticket to discuss this: https://github.com/cms-l1t-offline/cmssw/issues/913, maybe I misunderstood your code.

Cheers,
Dinyar